### PR TITLE
collapsible program documents

### PR DESCRIPTION
**Pull request by adamstankiewicz**
[ENT-1922](https://openedx.atlassian.net/browse/ENT-1922)

This PR adds collapsible functionality to the `Links` component. If more than 5 links are provided, we limit to only show the first 5 links and include a "Show all XX" button that, when clicked, renders the remaining links in the list.

### CODE REVIEW

File: src/components/DashboardHome/Sidebar.jsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import linksData from './sampleLinks';
 
 const Sidebar = () => (
   <>
-    <Links title={linksData.title} links={linksData.links} />
+    <Links id={linksData.id} title={linksData.title} links={linksData.links} />
```
Comment (muhammad-ammar) [inline src/components/DashboardHome/Sidebar.jsx]: I am not sure why we need this required `id` prop?

File: src/components/Links/Links.test.jsx [hunk 2]
```diff
@@ -1,33 +1,35 @@
...
-      .toJSON();
-    expect(tree).toMatchSnapshot();
+  let wrapper;
+
+  beforeEach(() => {
+    wrapper = shallow((
+      <Links id={linksData.id} title={linksData.title} links={linksData.links} />
+    ));
+  });
+
```
Comment (muhammad-ammar) [inline src/components/Links/Links.test.jsx]: Do we need a test where we have less than 5 links?

Also in the below tests, do we need to compare `show less`/`show all ...` button texts?

File: src/components/DashboardHome/Sidebar.jsx (hunk 1 above)
Comment (adamstankiewicz) [reply src/components/DashboardHome/Sidebar.jsx]: @muhammad-ammar For a11y, the `Links` component has an `aria-controls` attribute on the "show less / show all" button that needs to refer to an element with a unique id. I'm adding this `id` prop as way to pass in a unique value to the component if/when there are multiple `Links` components on the page.

File: src/components/Links/Links.test.jsx (hunk 2 above)
Comment (adamstankiewicz) [reply src/components/Links/Links.test.jsx]: I can add assertions to verify the button text for the "show less / show all" button as well as a test for less than 5 links.

**Pull request merged**